.github/workflows/e2e-test.yml

```
@@ -41,3 +41,13 @@ jobs:
       - name: Run E2E Tests ${{ matrix.config.name }}
         working-directory: ${{ matrix.config.folder }}
         run: make e2e-test
+
+      - name: Create reports ${{ matrix.config.name }}
+        working-directory: ./.github/workflows
+        run: ./create_reports.sh
+
+      - name: Upload ${{ matrix.config.name }} cluster snapshot 
+        uses: actions/upload-artifact@v3
```

You need to add `if: always()` here to tell github actions to also run those steps when the previous tests failed
```suggestion
      - name: Create reports ${{ matrix.config.name }}
        if: always()
        working-directory: ./.github/workflows
        run: ./create_reports.sh

      - name: Upload ${{ matrix.config.name }} cluster snapshot 
        if: always()
        uses: actions/upload-artifact@v3
```